src/inmanta/agent/executor.py

```diff
@@ -156,15 +162,18 @@ def from_specs(cls, code: typing.Collection["ResourceInstallSpec"]) -> "Executor
         sources = list({source for cd in code for source in cd.blueprint.sources})
         requirements = list({req for cd in code for req in cd.blueprint.requirements})
         pip_configs = [cd.blueprint.pip_config for cd in code]
+        python_versions = [cd.blueprint.python_version for cd in code]
         if not pip_configs:
             raise Exception("No Pip config available, aborting")
         base_pip = pip_configs[0]
+        python_version = python_versions[0]
```

Comment:
This relies on the same assumption as the line above, yet we don't assert it anywhere.